### PR TITLE
New "Learn More" on model metrics as metadata

### LINKED ISSUE #21: Add examples of preprocessing + model in "Learn more"
We scaled down the "Getting Started" example, so we should add back in some examples for showing how to deploy a pipeline/workflow involving both feature engineering + modeling.

### PR DESCRIPTION
**Pull request by juliasilge**
Well @isabelizimm after working on this a bit, I've decided I'm just going to ask you to add in the nicest ways to do some of this in Python. 😆  I'll note where we need your expertise!

### CODE REVIEW

File: learn-more/metrics-metadata.qmd [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+:::
+
+Now that our model is trained, we can estimate the model performance we expect to see on new data using our testing data.
+
+::: {.panel-tabset group="language"}
+## Python
+
+```{python}
+## predict on test data and compute metrics
+## possible to do accuracy, kappa, log loss, roc auc into one dataframe?
```
Comment (juliasilge) [inline learn-more/metrics-metadata.qmd:76]: What's the nicest way to predict on the testing data and compute some metrics so the results are in one dataframe? The defaults for tidymodels are these four but I can change which ones we use easily.

File: learn-more/metrics-metadata.qmd [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+## Python
+
+```{python}
+import vetiver
+
+v = vetiver.VetiverModel(
+    rf_pipe, 
+    prototype_data = X_train, 
+    model_name = "hotel-rf",
+    metadata = hotel_metrics.to_dict()
```
Comment (juliasilge) [inline learn-more/metrics-metadata.qmd:109]: Is this the best option? 

The way we do it in R will let folks put multiple things in this slot and get them out pretty nicely/easily, like `metadata = list(metrics = hotel_metrics, another_element = 1:10, more_custom_metadata = "potato")`.

File: learn-more/metrics-metadata.qmd [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+So far, we have walked through how to store metadata, but how do we extract our metrics out to use them?
+
+You can use `pin_meta()` to retrieve metadata from your board. All custom metadata is stored in a `"user"` slot; remember that other metadata is also automatically stored for you as well.
+
+::: {.panel-tabset group="language"}
+## Python
+
+```{python}
+metadata = model_board.pin_meta("hotel-rf")
+## Nicest way to get metrics dataframe back here?
```
Comment (juliasilge) [inline learn-more/metrics-metadata.qmd:170]: What's the best way to get back to a dataframe here?

File: learn-more/metrics-metadata.qmd [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+Use the `version` argument to `pin_meta()` to get the metadata for a specific model version.
+:::
+
+If you already have your vetiver model available, you can also retrieve the metadata directly:
+
+::: {.panel-tabset group="language"}
+## Python
+
+```{python}
+## Nicest way to get metrics back out of vetiver object?
```
Comment (juliasilge) [inline learn-more/metrics-metadata.qmd:191]: Again, what's the best way to get back to a dataframe from what's stored on the `v` object?

File: learn-more/metrics-metadata.qmd [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+df["arrival_date"] = pd.to_datetime(df["arrival_date"])
+df["arrival_month"] = df["arrival_date"].dt.month
+df["arrival_dow"] = df["arrival_date"].dt.dayofweek
+df = df.drop(columns="arrival_date").dropna()
+X, y = df.drop(columns="children"), df["children"]
+X_train, X_test, y_train, y_test = model_selection.train_test_split(
+    X, y, test_size=0.25
+)
+
+le = preprocessing.OrdinalEncoder().fit(X_train)
```
Comment (isabelizimm) [inline learn-more/metrics-metadata.qmd:36]: Right now, this is using the ordinal encoder on every column-- was this intentional, or should it just be for a subset of features?

Comment (isabelizimm) [reply learn-more/metrics-metadata.qmd:36]:
```
categorical_features = X_train.columns \
    .drop(['lead_time', 'stays_in_weekend_nights', 'stays_in_week_nights', 'adults', 'average_daily_rate', 'total_of_special_requests']).to_list()

oe = compose.make_column_transformer((preprocessing.OrdinalEncoder(), categorical_features), remainder = "passthrough").fit(X_train)
```

something like this will use the encoder on the categorical features and passthrough all the other features as-is.

Comment (juliasilge) [reply learn-more/metrics-metadata.qmd:36]: Ah yep! I was looking at a very old version of this model and just copied it over without looking in much detail. 🙈 

Are you able to push to this branch? Would you be up for doing a round of edits through the Python code?

File: learn-more/metrics-metadata.qmd [hunk 6]
```diff
@@ -0,0 +1,297 @@
...
+    period = td, 
+    metric_set = metric_set, 
+    truth = "children", 
+    estimate = "preds"
+)
+
+fig = vetiver.plot_metrics(td_metrics)
+
+# this does not assert preservation
+# AAA r2 and log loss are swapped on the plot??!?
```
Comment (isabelizimm) [inline learn-more/metrics-metadata.qmd:274]: Need to dig into this a little deeper-- right now order of metric is not preserved, so the log_loss and r2_score are appearing onto the wrong facet.

File: learn-more/metrics-metadata.qmd [hunk 7]
```diff
@@ -0,0 +1,297 @@
...
+
+Now that our model is trained, we can estimate the model performance we expect to see on new data using our testing data.
+
+::: {.panel-tabset group="language"}
+## Python
+
+```{python}
+from sklearn import metrics
+
+metric_set = [metrics.accuracy_score, metrics.log_loss, metrics.r2_score]
```
Comment (isabelizimm) [inline learn-more/metrics-metadata.qmd:103]: I grabbed this metric set, but it can be any [from this list](https://scikit-learn.org/stable/modules/classes.html#classification-metrics) that has a y_pred and y_true argument!

File: learn-more/metrics-metadata.qmd (hunk 4 above)
Comment (isabelizimm) [reply learn-more/metrics-metadata.qmd]: Added!

File: learn-more/metrics-metadata.qmd (hunk 3 above)
Comment (isabelizimm) [reply learn-more/metrics-metadata.qmd]: Added!

File: learn-more/metrics-metadata.qmd (hunk 2 above)
Comment (isabelizimm) [reply learn-more/metrics-metadata.qmd]: Yep! to_dict() is the right way to go.

File: learn-more/metrics-metadata.qmd (hunk 1 above)
Comment (isabelizimm) [reply learn-more/metrics-metadata.qmd]: I grabbed this metric set, but it can be any [from this list](https://scikit-learn.org/stable/modules/classes.html#classification-metrics) that has a y_pred and y_true argument!

**Comment by juliasilge** [comment]
Looking great! I was able to render the whole thing (because miraculously I _do_ have working Python envs 😆) and it looks good.
